db/commitlog/commitlog.cc

```diff
@@ -1031,6 +1033,8 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
                 }
             });
 
+            co_await coroutine::switch_to(_segment_manager->cfg.sched_group);
+
```

Comment:
Why here, of all places?

I don't know that it's wrong, but how did you choose it? Just before I/O?

Ah, you wrote so. I don't know if that's the best strategy, but we can adjust it later.

I hope we don't get wierd schedulingr_group leaks from the unbalanced switch_to.

Reply:
> Why here, of all places?
> 
> I don't know that it's wrong, but how did you choose it? Just before I/O?> 

IO needs to happen in the context of this new sched group, so it's "around" dma_write-s.

In fact, I first put the switch earlier causing more code from segment::cycle() run in it, but it looks like cycle() entrance through this loop should be non-preemptive. Otherwise `_buffer`  gets smashed and scylla throws/asserts on buffer vs entries checks. E.g. like this

```
[('node1', ["scylla: db/commitlog/commitlog.cc:1090: future<db::commitlog::segment::sseg_ptr> db::commitlog::segment::batch_cycle(timeout_clock::time_point): Assertion `_segment_manager->cfg.mode != sync_mode::BATCH || _flush_pos > fp' failed.", 'Aborting on shard 1.'])]
```

or

```
[shard 1] storage_proxy - exception during mutation write to 127.0.13.2: utils::internal::nested_exception<std::runtime_error> (Could not write mutation system:paxos (pk{000774657374333838}) to commitlog): std::invalid_argument (Attempting to insert key out of order: {1, 556792, 25440256})'])]"
```

I didn't dig it deep enough to describe exactly what's going on